libraries/Microsoft.Bot.Connector/Authentication/BuiltinCloudEnvironment.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    internal abstract class BuiltinCloudEnvironment : ICloudEnvironment
+    {
+        private string _toChannelFromBotOAuthScope;
+        private string _loginEndpoint;
+        private string _callerId;
+        private string _channelService;
+        private IServiceClientCredentialsFactory _credentialFactory;
+        private AuthenticationConfiguration _authConfiguration;
+        private HttpClient _httpClient;
+        private ILogger _logger;
+
+        public BuiltinCloudEnvironment(string toChannelFromBotOAuthScope, string loginEndpoint, string callerId, string channelService, IServiceClientCredentialsFactory credentialFactory, AuthenticationConfiguration authConfiguration, HttpClient httpClient, ILogger logger)
+        {
+            _toChannelFromBotOAuthScope = toChannelFromBotOAuthScope;
+            _loginEndpoint = loginEndpoint;
+            _callerId = callerId;
+            _channelService = channelService;
+            _credentialFactory = credentialFactory;
+            _authConfiguration = authConfiguration;
+            _httpClient = httpClient;
+            _logger = logger;
```

Comment:
Should we default to default logger or throw if null?

Reply:
You're right, I think default to logger.null or whatever the class is called is the correct behavior